managed/src/main/java/com/yugabyte/yw/controllers/AccessKeyController.java

```diff
@@ -165,20 +143,20 @@ public Result delete(UUID customerUUID, UUID providerUUID, String keyCode) {
       Audit.createAuditEntry(ctx(), request());
       return ApiResponse.success("Deleted KeyCode: " + keyCode);
     } else {
-      return ApiResponse.error(INTERNAL_SERVER_ERROR, "Unable to delete KeyCode: " + keyCode);
+      throw new YWServiceException(INTERNAL_SERVER_ERROR, "Unable to delete KeyCode: " + keyCode);
     }
   }
 
   private String validateUUIDs(UUID customerUUID, UUID providerUUID) {
```

Comment:
See if there is existing method in Provider or make one getOrBadRequest(customerUUID, providerUUID)